lib/manageiq/appliance_console/database_configuration.rb

```diff
@@ -46,7 +46,9 @@ def run_interactive
       post_activation
       say("\nConfiguration activated successfully.\n")
     rescue RuntimeError => e
-      puts "Configuration failed#{": " + e.message unless e.class == MiqSignalError}"
+      if e.class != MiqNothingChanged
```

Comment:
Also, don't do this.

If we can't rescue just the exception type we care about then we need to find a better way to not print an error message. Depending on exceptions to implement what is essentially a `goto` is bad enough without doing explicit class type comparisons.

Reply:
Thanks for this useful hint. Part of your comment I'm not understood. What does `without doing explicit class type comparisons` means?

Reply:
You're comparing the class of the error raise to another class within a rescue block.

This really shouldn't be necessary and this kind of thing usually means that the rest of the code could use improvement/refactoring to remove the need for it.

Reply:
I see. I think i should return a third value for `activate` to indicate nothing changes and check what it returns instead of `raise MiqSignalError unless activate`. But as you discussed with Yuri, you're agree it's misunderstanding. So I'll revert this (and still show `Configuration failed`), right?

Reply:
Yes, I think that the only change for this bug should be the hint about how to remove the database connections.